c7n/resources/quicksight.py

```diff
@@ -141,7 +133,12 @@ def resources(self, query):
         required = {
             "AwsAccountId": self.manager.config.account_id
         }
-        return super().resources(required)
+        try:
+            required_resources = super().resources(required)
+        except ClientError as e:
+            if is_quicksight_account_missing(e):
+                return []
```

Comment:
This should re raise if nots in the proscribed set, else its silently swallowing the error and returning an undefined variable.

Reply:
Good catch! I forgot to add back an else there to raise exception

Reply:
added `else: raise` in two of the places it was missing from.